pkg/webhook/policy.go

```diff
@@ -181,24 +180,34 @@ func (h *validationHandler) Handle(ctx context.Context, req admission.Request) a
 	}
 
 	res := resp.Results()
-	msgs := h.getDenyMessages(res, req)
+	msgs, enforcementAction := h.getDenyMessages(res, req)
 	if len(msgs) > 0 {
-		vResp := admission.ValidationResponse(false, strings.Join(msgs, "\n"))
+		var vResp admission.Response
 		if vResp.Result == nil {
 			vResp.Result = &metav1.Status{}
 		}
-		vResp.Result.Code = http.StatusForbidden
-		requestResponse = denyResponse
+
+		if enforcementAction == string(util.Warn) {
+			vResp = admission.ValidationResponse(true, strings.Join(msgs, "\n"))
```

Comment:
Have we tested that this works as expected on K8s clusters prior to 1.19?

Reply:
Yes, it's same behavior as using dryrun.